golang/publisherConnector/main.go

```diff
@@ -0,0 +1,354 @@
+							},
+						},
+						{
+							Key: helpers.PROPERTY_KEY_COLOR,
+							Value: &ioticsApi.Property_StringLiteralValue{
+								StringLiteralValue: &ioticsApi.StringLiteral{
+									Value: "#9aceff",
+								},
+							},
+						},
+						{
+							Key: helpers.PROPERTY_KEY_CREATED_BY,
+							Value: &ioticsApi.Property_StringLiteralValue{
+								StringLiteralValue: &ioticsApi.StringLiteral{
+									Value: "Lorenzo Paris",
+								},
+							},
+						},
+						// Saref TemperatureSensor Ontology
+						// Some of the properties of this ontology can be used to add more description to the Sensor
+						{
+							Key: helpers.PROPERTY_KEY_DEFINES,
+							Value: &ioticsApi.Property_UriValue{
+								UriValue: &ioticsApi.Uri{
+									Value: helpers.PROPERTY_VALUE_SAREF_TEMPERATURE_SENSOR,
+								},
+							},
+						},
+						// Add "hasModel" property
+						{
+							Key: helpers.PROPERTY_KEY_SAREF_HAS_MODEL,
+							Value: &ioticsApi.Property_StringLiteralValue{
+								StringLiteralValue: &ioticsApi.StringLiteral{Value: "T1234"},
+							},
+						},
+						{
+							Key: helpers.PROPERTY_KEY_HOST_METADATA_ALLOW_LIST,
+							Value: &ioticsApi.Property_UriValue{
+								UriValue: &ioticsApi.Uri{
+									Value: helpers.PROPERTY_VALUE_ALLOW_ALL_HOSTS,
+								},
+							},
+						},
+						{
+							Key: helpers.PROPERTY_KEY_HOST_ALLOW_LIST,
+							Value: &ioticsApi.Property_UriValue{
+								UriValue: &ioticsApi.Uri{
+									Value: helpers.PROPERTY_VALUE_ALLOW_ALL_HOSTS,
+								},
+							},
+						},
+					},
+					Feeds: []*ioticsApi.UpsertFeedWithMeta{
+						{
+							Id: "temperature",
+							Properties: []*ioticsApi.Property{
+								{
+									Key: helpers.PROPERTY_KEY_LABEL,
+									Value: &ioticsApi.Property_LangLiteralValue{
+										LangLiteralValue: &ioticsApi.LangLiteral{
+											Value: "Temperature",
+											Lang:  "en",
+										},
+									},
+								},
+								{
+									Key: helpers.PROPERTY_KEY_COMMENT,
+									Value: &ioticsApi.Property_LangLiteralValue{
+										LangLiteralValue: &ioticsApi.LangLiteral{
+											Value: "Current temperature of a Room",
+											Lang:  "en",
+										},
+									},
+								},
+							},
+							Values: []*ioticsApi.Value{
+								{
+									Label:    "reading",
+									Comment:  "Temperature in Celsius degrees",
+									DataType: "integer",
+									Unit:     helpers.UNIT_DEGREE_CELSIUS,
+								},
+							},
+						},
+					},
+					Location: &ioticsApi.GeoLocation{
+						Lat: 51.864, Lon: -0.412,
+					},
+				},
+			},
+		)
+		if err != nil {
+			log.Fatalf("Could not Upsert Twin: %v", err)
+		}
+		log.Printf("Created Twin %s", twinTemperatureIdentity.Did())
+
+		// Update mapping
+		roomNumberToTwinId[roomNumber] = twinTemperatureIdentity.Did()
+	}
+
+	/*** SHARE DATA ***/
+	/* For each Twin we want to generate a random tenmperature and share the data */
+	for {
+		for roomNumber := range roomNumberToTwinId {
```

Comment:
and here you could just do
`for twinId := range twinIds`
and don't have to use the roomNumber index.